pyteal/ast/txn.py

```diff
@@ -728,6 +751,20 @@ def state_proof_pk(self) -> TxnExpr:
         """
         return self.makeTxnExpr(TxnField.state_proof_pk)
 
+    def num_approval_program_pages(self) -> TxnExpr:
+        """Get the number of pages in the approval program.
+
+        Requires TEAL version 7 or higher.
+        """
+        return self.makeTxnExpr(TxnField.num_approval_program_pages)
+
+    def num_clear_state_program_pages(self) -> TxnExpr:
+        """Get the number of pages in the clear state program.
+
+        Requires TEAL version 7 or higher.
+        """
+        return self.makeTxnExpr(TxnField.num_clear_state_program_pages)
```

Comment:
The convention is to not define array length fields. Instead, `Txn. approval_program_pages.length()` can be used

Reply:
@jasonpaulos I attempted to address in https://github.com/algorand/pyteal/pull/452.  Can you have a look and resolve if the PR captures it?